app/controllers/posts_controller.rb

```diff
@@ -9,4 +9,27 @@ def show
     @post = Post.find(params[:id])
     @user = @post.author
   end
+
+  def new
+    @post = Post.new
+    @user = User.find(params[:user_id])
+  end
+
+  def create
+    @post = Post.new(post_params)
+    @user = User.find(params[:user_id])
+    @post.author = @user
```

Comment:
- [ ] Posts should be created on behalf of the `current_user`, not the user from the URL param. Kindly fix this in your create method.